stdlib.sh

```diff
@@ -1068,6 +1068,33 @@ layout_pyenv() {
   [[ -n "$PYENV_VERSION" ]] && export PYENV_VERSION
 }
 
+# Usage: layout uv
+#
+# Enables the uv project layout in the current directory, and syncs
+# the dependencies in the project.
+#
+# This relies on the `uv` command being available in the PATH, and performs a
```

Comment:
I am not sure addinc `uv sync` is great. It might be fine for smaller repos, but for larger ones it could be annoying.

Personally I have git hooks that one rebase/merge checks the applicable files and only then syncs. This is much more intrusive then when I jump back/forth between folders and it syncs each time.

I think we should keep `layout uv` to a minimum of activating the venv.

Reply:
> I think we should keep `layout uv` to a minimum of activating the venv.

But call it `use venv` or something like that? Because then it doesn't have any uv-specific features, does it?

Reply:
I am not sure I agree. It seems that the Python specific `layout_` has unfortunately gone on a path to do installation/syncing. But if you look at most layouts, they do not do that. It is intrusive in a way that slows down the prompt.

The description of `layout <type>` even to me indicates it should not do things like installations such as with `uv sync`:

> A semantic dispatch used to describe common project layouts.

@zimbatm what is your take on this?

Reply:
You seem to assume that I am against leaving sync out (but my interpretation might be wrong?). I am not. I am merely saying "don't call it uv if it isn't uv-specific".

Reply:
Your interpretation was correct. I thought you were against leaving sync out.

Since you would/should most likely use `UV_PROJECT_ENVIRONMENT` for the virtual env path it would still be uv specific 🙂

Reply:
I think the value of this would be essentially nil without the sync + watch parts; it's just another location for a virtualenv, then, and that's a one-liner in any `.envrc` anyway.

As far as speed goes, `uv` is faster than nearly any other part of my environment setup, regardless of how complex a project I've thrown at it. Obviously I cannot predict how that performance will change over time, but the trends are positive.

I'd say that if syncing isn't desirable here -- I believe it is, but that's definitely up for debate! -- then this is probably not the ideal change to make to the stdlib.

As a personal anecdote, this is how I've been running direnv for a couple of months; what's in this PR is a cleaned up and streamlined version of my own direnv lib for uv Python, and it is *seamless*; I think my prompt generator takes more time than this does.

Reply:
@offbyone it is not fast if you need to download packages etc. faster than anything else surely, but not immediate. It becomes subjective what fast is if it is left there. It is also more intrusive since it does on-disc changes and not just _environment_ changes.

Reply:
Of course. That's why I didn't propose integrating it into the existing `layout python` suite; this is a different layout, with different expectations, that is entirely optional to use. People who _do_ use it will get what they expect, and those who don't won't be affected.

I am proposing this for the stdlib because I think it's a very useful and ergonomic pattern, and I'd like to use it out of the stock direnv, but if it's not the direction you want to take it in, then that's also fine too. 

Is there some kind of documentation change I could add to this that would make the expectations clear in a way that would make merging it reasonable?

Reply:
If a `layout uv` is added, we should strive for a smallest common denominator at least I think. Saying it is optional is not fair, it is optional for you as well in that sense to not use it.

Or rather:
The benefit of not adding it into this layout is that anyone, who want optionality, can add their own `watch_file` and `uv sync` in their own `.envrc`. But if `uv sync` is left in, it is added not optional for others.

I understand you want this from the vantage point of your workflow. I am not sure everyone agrees with that workflow though.

(I am not 100% opposed to having it in there, but I am not sure it is the intention of layouts to do such things from the start).

Reply:
I'm not a nix user, but how is this different from `use flake`? Does that not download / build if it's not already cached?